projects/com.oracle.truffle.llvm.types/src/com/oracle/truffle/llvm/types/memory/LLVMStack.java

```diff
@@ -76,6 +76,10 @@ public void free() {
         isFreed = true;
     }
 
+    public static long allocate(long size) {
```

Comment:
Actually, since signal handler support does not need to be fast (yet), we can also allocate the memory once we enter the signal handler and deallocate it again once we leave it.